server/edd/load/parsers/core.py

```diff
@@ -641,6 +645,127 @@ def _raw_cell_value(self, cell):
         return val
 
 
+class MultiSheetExcelParserMixin:
+    def parse(self, file):
+        """
+        Parses the input as a Pandas dataframe and then
+        converts it into an Excel workbook.
+
+        :param file: a file-like object
+        :return: the ParseResult read from file, otherwise None
+        :raises OSError: if the file can't be opened
+        :raises EDDImportError: if the file format or content is bad
+        """
+
+        wb = pd.read_excel(file, sheet_name=None)
+
+        # for each worksheet in the workbook
+        parsed_result = pd.DataFrame()
+        mapper = MeasurementMapper()
+
+        for name, sheet in wb.items():
+            # for every two columns in the worksheet
+            # corresponding to each measurement type in the sheet
+            for i in range(0, int(sheet.shape[1]), 2):
+                two_cols = sheet[sheet.columns[i : i + 2]]
+
+                # dropping all rows with nan values in the worksheet
+                two_cols = two_cols.dropna()
+                if not two_cols.dropna().empty:
+                    # decimate the data
+                    two_cols = two_cols.iloc[::10, :]
+
+                    # using mapper to map data into the EDD import format
+                    # and convert in to a pandas dataframe
+                    # set the line name and the dataframe with the two columns
+                    # with data for the next measurement type
+                    mapper.set_line_name(name)
+                    mapper.set_data(two_cols)
+                    parsed_df = mapper.map_data()
+                    parsed_result = parsed_result.append(parsed_df)
+
+        # convert mapper.df into a openpyxl worksheet
+        # NOTE: work on using openpyxl for the manipulation
+        # in the mapper instead of pandas
+        wb = Workbook()
+        ws = wb.active
+        for r in dataframe_to_rows(parsed_result, index=True, header=True):
+            ws.append(r)
+
+        # passing the rows in the worksheet for verification
+        # and processing in database
+        return self._parse_rows(ws.iter_rows())
+
+    def _raw_cell_value(self, cell):
+        """
+        Gets the raw cell value in whatever format it was stored in the file.
+
+        :param cell: the cell
+        :return: the cell value, with leading and trailing whitespace stripped
+            if the content was a string
+        """
+        val = cell.value
+        if isinstance(val, str):
+            return val.strip()
+        return val
+
+
+class MeasurementMapper:
+
+    loa_name: str
+    parsed_df: pd.DataFrame
+    df: pd.DataFrame
+    units: Dict
+    mes_unit_map: Dict
+
+    def __init__(self, name=None, df=None):
+        self.loa_name = name
+        self.df = df
+        # maps naming format of Ambr measurement type to
+        # naming convention in EDD
+        self.mtypes = {
```

Comment:
Don't hard-code these. If the input file uses different names than what is currently in EDD, that translation needs to happen at the data level, and not as hard-coded entries.